### DISCUSSION TITLE
CoPilot parse .gitignore and NOT enable it on matching files

### DISCUSSION THREAD

**Discussion by nodesocket** [community/community#51183] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

This is **very concerning** as passwords, secrets, and api keys are being sent up over the internet and stored on OpenAI.

It seems obvious that CoPilot should parse `.gitignore` just like git does and dynamically add those files to exclusion of CoPilot unless you explicitly opt-in.

**Comment by hochoy** [comment]
actually, have a look here. There are other ways to exclude files from copilot, but yea, it's not (yet) as flexible as `.gitignore`.

https://github.com/orgs/community/discussions/11254
https://github.com/orgs/community/discussions/13334

**Comment by Plarixtu** [comment]
I just wanted to try Copilot but because of this I will not be using it. I'm only using it within IDE and not on a repository that is hosted on Github. So I was not able to find an option to exclude my secret environment files. They are all excluded by .gitignore on purpose.
